### PR TITLE
Tests and Documentation for FluentNavigators and RootFrameNavigationPageViewModels

### PR DESCRIPTION
**Pull request by SavikPavel**
### Description

- Tests and Documentation added for FluentNavigators and RootFrameNavigationPageViewModels

### API Changes
 
 None

### Platforms Affected

- Core (all platforms)

### Behavioral/Visual Changes

None

### Before/After Screenshots

Not applicable

### PR Checklist

- [x] I have read the [CONTRIBUTING](https://github.com/Softeq/XToolkit.WhiteLabel/blob/master/.github/CONTRIBUTING.md) document
- [x] My code follows the [code styles](https://github.com/Softeq/dotnet-guidelines)
- [x] Targets the correct branch
- [x] Tests are passing (or failures are unrelated)

### CODE REVIEW

File: Softeq.XToolkit.WhiteLabel/Navigation/FluentNavigators/FluentNavigatorExtensions.cs [hunk 1]
```diff
@@ -9,10 +10,30 @@ namespace Softeq.XToolkit.WhiteLabel.Navigation.FluentNavigators
...
+        /// </summary>
+        /// <param name="viewModel">ViewModel to add parameters to.</param>
+        /// <param name="parameters">List of navigation parameters to add.</param>
+        /// <exception cref="T:System.ArgumentNullException">
+        ///     <paramref name="viewModel"/>
+        ///     and <paramref name="parameters"/> cannot be <see langword="null"/>.
+        /// </exception>
+        public static void ApplyParameters(
+            this IViewModelBase viewModel,
             IEnumerable<NavigationParameterModel>? parameters)
```
Comment (wcoder) [inline Softeq.XToolkit.WhiteLabel/Navigation/FluentNavigators/FluentNavigatorExtensions.cs:24]: "?" can be removed

File: Softeq.XToolkit.WhiteLabel/Navigation/FluentNavigators/FrameFluentNavigator.cs [hunk 2]
```diff
@@ -8,42 +8,103 @@
...
+        ///     Initializes current service with the specified <paramref name="initParameter"/>.
+        /// </summary>
+        /// <param name="initParameter">
+        ///      Navigation object (usually platform-specific) that is used for initialization.
+        /// </param>
+        /// <returns>Self.</returns>
         public FrameFluentNavigator<TViewModel> Initialize(object initParameter)
         {
-            _frameNavigationService.Initialize(initParameter);
+            _navigationService.Initialize(initParameter);
```
Comment (wcoder) [inline Softeq.XToolkit.WhiteLabel/Navigation/FluentNavigators/FrameFluentNavigator.cs:85]: Seems, need null-heck for `initParameter`

File: Softeq.XToolkit.WhiteLabel.Tests/Mvvm/RootFrameNavigationPageViewModelTests/RootFrameNavigationPageViewModelTests.cs [hunk 3]
```diff
@@ -0,0 +1,73 @@
...
+            _frameNavigationService.Received(1).NavigateToViewModel<ViewModelStub>(
+                Arg.Is(true),
+                Arg.Is((IReadOnlyList<NavigationParameterModel>) null));
+        }
+
+        [Fact]
+        public void InitializeNavigation_InitializesNavigationService()
+        {
+            var param = new object();
+            _vm.InitializeNavigation(param);
```
Comment (wcoder) [inline Softeq.XToolkit.WhiteLabel.Tests/Mvvm/RootFrameNavigationPageViewModelTests/RootFrameNavigationPageViewModelTests.cs:58]: Please add test for `null` case

File: Softeq.XToolkit.WhiteLabel.Tests/Navigation/FluentNavigators/FluentNavigatorExtensionsTests/FluentNavigatorExtensionsTests.cs [hunk 4]
```diff
@@ -0,0 +1,66 @@
...
+
+            Assert.Throws<ArgumentNullException>(() => viewModel.ApplyParameters(parameters));
+        }
+
+        [Fact]
+        public void ApplyParameters_CalledOnNonNull_WithNullParams_ThrowsCorrectException()
+        {
+            IViewModelBase viewModel = new ViewModelStub();
+
+            Assert.Throws<ArgumentNullException>(() => viewModel.ApplyParameters(null));
```
Comment (wcoder) [inline Softeq.XToolkit.WhiteLabel.Tests/Navigation/FluentNavigators/FluentNavigatorExtensionsTests/FluentNavigatorExtensionsTests.cs:38]: Case:
```cs
var p = new List<NavigationParameterModel>
{
    null!,
    null!
};
```

File: Softeq.XToolkit.WhiteLabel/Navigation/FluentNavigators/FrameFluentNavigator.cs (hunk 2 above)
Comment (SavikPavel) [reply Softeq.XToolkit.WhiteLabel/Navigation/FluentNavigators/FrameFluentNavigator.cs:93]: Fixed

File: Softeq.XToolkit.WhiteLabel/Navigation/FluentNavigators/FluentNavigatorExtensions.cs (hunk 1 above)
Comment (SavikPavel) [reply Softeq.XToolkit.WhiteLabel/Navigation/FluentNavigators/FluentNavigatorExtensions.cs]: Fixed

File: Softeq.XToolkit.WhiteLabel.Tests/Mvvm/RootFrameNavigationPageViewModelTests/RootFrameNavigationPageViewModelTests.cs (hunk 3 above)
Comment (SavikPavel) [reply Softeq.XToolkit.WhiteLabel.Tests/Mvvm/RootFrameNavigationPageViewModelTests/RootFrameNavigationPageViewModelTests.cs:64]: Fixed

**Review by wcoder — APPROVED** [review]
